### ISSUE TITLE
Creating new datastore blocks log in

### ISSUE THREAD

**Issue by larrysalibra** [blockstack/blockstack-storage-js#3]
It looked like log in had failed or nothing happened for about ~10 seconds after being redirected to the hello blockstack for the nth time app.

Making a note of this for the moment:

```
Navigated to http://localhost:5000/
bundle.js:673 No datastore contexts defined
bundle.js:777 Driver dropbox implementes read_public
bundle.js:777 Driver disk implementes read_private
bundle.js:777 Driver disk implementes write_private
bundle.js:777 Driver dropbox implementes write_private
bundle.js:777 Driver disk implementes read_local
bundle.js:777 Driver disk implementes write_local
bundle.js:853 Driver dropbox fulfills replication concern public
bundle.js:862 Select driver dropbox
bundle.js:853 Driver disk fulfills replication concern local
bundle.js:862 Select driver disk
bundle.js:1068 Will use drivers dropbox,disk
bundle.js:1069 Datastore will span devices 0
1KVUb4wKvCC2qufvXpSYgbPK3QnnB2vKWL:1 GET http://localhost:6270/v1/stores/1KVUb4wKvCC2qufvXpSYgbPK3QnnB2vKWL?device_ids=0&blockchain_id=person.id 404 (Not Found)
bundle.js:1079 Datastore does not exist; creating...
null:1 GET http://localhost:5000/null 404 (Not Found)
app.js:75 Saved! Took 9.975 seconds
```

**Comment by jcnelson** [comment]
Is the specific problem that the messages take ~10 seconds to show up?  That's due to slowness in the Dropbox driver.  I have some optimization fixes in Core rc-0.14.3 that bring this down to ~2 seconds.

**Comment by larrysalibra** [comment]
I guess the answer is that apps processing sign in should have some sort of “please wait” indicator while waiting for the `handlePendingSignIn()` Promise to resolve.
